codestyle/styles.md

```diff
@@ -0,0 +1,672 @@
+* it is impossible to use such blocks on the page more than one times;
+
+**No selectors by element**
+
+```CSS
+/* Wrong: */
+
+p {
+    margin: 1em 0;
+}
+
+.block img {
+    border: 1px solid #000;
+}
+```
+*Reasons:*
+* because with high probability the rules will be applied to elements you were
+  not going apply to;
+* inability to replace one tag on another without changing styles;
+
+*Exceptions:*
+* allowed to use in the reset/normalize styles files, which set basic
+  styles for the whole project;
+* allowed to use for styling selectors related to tables;
+
+**No selectors by attribute**
+
+```CSS
+/* Wrong: */
+
+.input[type="text"] {
+    font-size: 1em;
+}
+
+.input[disabled] {
+    opacity: 0.5;
+}
+
+
+/* Right: */
+
+.input_type_text {
+    font-size: 1em;
+}
+
+.input_state_disabled {
+    opacity: 0.5;
+}
+```
+
+*Reasons:*
+* the attribute selector has the same specificity as the class selector which
+  leads to an increase in the specificity and complexity of the redefinition of
+  the rules;
+
+*Exceptions:*
+* in case we need to style embedded a third-party widgets we have no ability to
+  edit its markup we can use this approach:
+
+```CSS
+[id="third-party-widget"] {}
+```
+
+
+**No cascade selectors**
+
+*Reasons:*
+- ...
+
+*Exceptions:*
+- ...
+
+
+
+<a id="values"></a>
+## Property values
+
+<a id="values_numbers"></a>
+### Numbers
+* No leading zero. `.3` instead of `0.3`;
+* Maximum three (3) places after the point. `2.345em` is ok, but `2.3456` is
+  wrong;
+
+<a id="values_colors"></a>
+### Colors
+* 6-character color hash notation for color calues without transparency:
+  `#44B1C3`;
+* 3-character color hash notation everywhere it is possible. For example `#F60`
+  instead of `#FF6600`;
+* functional notation for colors with opacity different than one (1):
+  `rgba(34, 12, 64, 0.3)`;
+* colors in the hash notation are uppercased: `#44B1C3`. Not: `#44b1c3`;
+
+<a id="values_others"></a>
+### Other
+* values inside `url()` construction without quotes:
+
+```CSS
+.block {
+    background-image: url(/img/sprite.svg);
+}
+```
+
+* no prefixed properties (it does Autoprefixer);
+* no base64 encoded images (it does a builder);
+* no `!important` directive;
+  * Except cases when you use it proactively. Proactive use of `!important`
+    means that it's used as a guarantee that some rule always will work, but not
+    as a fix of a specificity problem you’ve already encountered. For example:
+
+    ```CSS
+        .hidden {
+            display: none !important;
+        }
+    ```
+
+* no units for zero values: `margin: 0` is ok, but: `margin: 0px` is wrong;
+
+
+
+<a id="bem"></a>
+## BEM
+
+Describing the essence of BEM methodology is beyond the scope of
+this document, everything you need you can find on the official BEM site:
+[BEM methodology](https://en.bem.info/method/). Here are just described the
+formal rules of naming CSS classes.
+
+<a id="bem_naming"></a>
+### Naming conventions
```

Comment:
I put it here because before we write CSS-classes we have to agree about basic things like indents, encoding and so on.